Sources/FoundationEssentials/String/String+Path.swift

```diff
@@ -665,7 +665,9 @@ extension String {
                 guard GetFinalPathNameByHandleW(hFile, $0.baseAddress, dwLength, VOLUME_NAME_DOS) == dwLength - 1 else {
                     return nil
                 }
-                return String(decodingCString: $0.baseAddress!, as: UTF16.self)
+
+                // When using `VOLUME_NAME_DOS`, the returned path uses `\\?\`.
+                return String(String(decodingCString: $0.baseAddress!, as: UTF16.self).dropFirst(4))
```

Comment:
```suggestion
                return String(decodingCString: $0.baseAddress!.advanced(by: 4), as: UTF16.self)
```

Are we able to advance the pointer here instead of calling `dropFirst` to avoid needing to create an intermediate `String` and copy to a new `String` (just a minor performance optimization)

Reply:
Yes, that is a good idea!